src/Helper/Helper.php

```diff
@@ -115,4 +144,53 @@ public static function castToCBCustomType( $post, $type ) {
 		return $post;
 	}
 
+	/**
+	 * Returns one or more overlapping timeframes, given an array of timeframes
+	 *
+	 * NOTE: When performance issues arise, this operation can be implemented
+	 * faster with an interval tree data structure
+	 *
+	 * @param $arrayOfRanges
+	 *
+	 * @return array():TimeFrame
+	 */
+	public static function mergeRangesToBookableDate( $arrayOfRanges ): array {
+
+		if ( count($arrayOfRanges) == 1) {
+			return $arrayOfRanges;
+		}
+
+		$result = array();
+
+		// Sort by start date
+		usort($arrayOfRanges, function( $a, $b ) {
+			return $a['start_date'] <=> $b['start_date'];
+		});
+
+		$result[] = $arrayOfRanges[0];
+		$last = 0;
+
+		// For each range, compare with last (or first) merged range
+		for ($i = 1; $i < count($arrayOfRanges); $i++) {
+
+			// Either
+			//  If first/last interval is open => overlaps next
+			//  Or first/last interval end is greater than next interval begin
+			if (
+				interval_open($result[ $last ]['end_date'])
+			    || $result[ $last ]['end_date'] >= $arrayOfRanges[$i]['start_date'])
+			{
+				// => Overlap, merge both
+				set_false_if_either_key_null_else_set_func( 'start_date', $result[ $last ], $arrayOfRanges[ $i ], function($a,$b) {return min($a, $b);} );
```

Comment:
Ich verstehe nicht ganz, warum 'start_date' auf false geprüft wird, soweit ich weiß kann nur end_date false sein, oder?

Reply:
Das stimmt. Das Frontend lässt nie eine Anlage ohne start_date zu. Falsch wäre es aber dennoch nicht (also auf offene Intervalle in beide Richtungen zu prüfen).

Ich glaube ich nehme da eh lieber `TimeFrame::hasTimeframeDateOverlap` ... dann ist es an einer Stelle zentral implementiert was ein Overlap bedeutet.

https://github.com/wielebenwir/commonsbooking/blob/master/src/Model/Timeframe.php#L489

Was denkst du?

Reply:
Klar, das Problem ist halt, dass du von dem Template nur die ranges bekommst, dann müsste man irgendwie wieder aufs Model zurückgreifen.

Reply:
Ok dann muss ich erstmal verstehen wie Daten ins Template gelangen (und warum sie dann kein Model mehr sind)

Reply:
@hansmorb ich habe den check für die offenen intervall am Beginn mit dem letzten commit rausgenommen.